code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -408,3 +493,251 @@
 	. = ..()
 	atom_storage.max_slots = 4
 	atom_storage.set_holdable(list(/obj/item/coffee_cartridge))
+
+
+
+/*
+ * impressa coffee maker
+ * its supposed to be a premium line product, so its cargo-only, the board cant be therefore researched
+ */
+
+/obj/machinery/coffeemaker/impressa
+	name = "impressa coffeemaker"
+	desc = "An industry-grade Impressa Modello 5 Coffeemaker of the Piccionaia Home Appliances premium coffeemakers product line. Makes coffee from fresh dried whole beans."
+	icon = 'icons/obj/machines/coffeemaker.dmi'
+	icon_state = "coffeemaker_impressa"
+	circuit = /obj/item/circuitboard/machine/coffeemaker/impressa
+	initial_cartridge = null		//no cartridge, just coffee beans
+	brew_time = 15 SECONDS			//industrial grade, its faster than the regular one
+	var/const/bean_capacity = 10	//how many beans can fit inside
```

Comment:
All of these new variables should be DMDOC'd; see https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md

Reply:
Both suggestions addressed in [cba0e96](https://github.com/tgstation/tgstation/pull/70991/commits/cba0e96ea015db96cd66e1eecf698269a6049a8e)